conda/core/index.py

```diff
@@ -93,10 +80,11 @@ def get_index(channel_urls=(), prepend=True, platform=None,
         channel_urls += context.channels
 
     channel_priority_map = prioritize_channels(channel_urls, platform=platform)
-    index = fetch_index(channel_priority_map, use_cache=use_cache, unknown=unknown)
 
-    if prefix:
-        supplement_index_with_prefix(index, prefix, channel_priority_map)
+    prefix_index = build_linked_data_index(prefix, channel_priority_map) if prefix else {}
+    index = fetch_index(channel_priority_map, use_cache=use_cache, unknown=unknown,
+                        index=prefix_index)
```

Comment:
It seems to me this simplifies the logic.  But please think through what I've removed from `supplement_index_with_prefix()` when translating it into `build_linked_data_index()`.  Please make sure I haven't removed anything that's important.